File: core/src/engine/rts/model/TroopCollection.scala
package engine.rts.model

/**
 * This class represents a collection of troops.
 */
class TroopCollection{
   var deathCount=0 //This is to keep track of the number of deaths
   
   def addDeath(){
       deathCount+=1
   }
   def resetDeathCount(){
       deathCount=0
   }
}

File: core/src/engine/rts/model/TroopCollection.java
package engine.rts.model;

public class TroopCollection {
	
   protected int deathCount=0;//This is to keep track of the number of deaths
			   
   public void addDeath(){
       deathCount+=1;
   }
   
   public void resetDeathCount(){
       deathCount=0;
   }
     
}